test_oop Ruby

Vali-Mandeal opened the pull request.

Vali-Mandeal committed:
xD

alexandru-calinoiu commented:
Please give a proper name to the pr and add a description to it with a minimal info on how to run the code, ruby version other deps ....

alexandru-calinoiu reviewed (changes requested):
Please give a proper name to the pr and add a description to it with a minimal info on how to run the code, ruby version other deps ....

Vali-Mandeal committed:
Proper Description

- The Ruby Version I used is:
	ruby 2.6.5p114 (2019-10-01 revision 67812) [x64-mingw32]

- I used Atom IDE 1.40.1 with platformio-ide-terminal 2.9.4 for taking user input
- No other dependencies needed.

In order to run the Project run the Main.rb file
	- With my setup just write this command in the terminal: "ruby CoffeeShopApp/Main.rb"
	- Provide your x location
	- Provide your y location
	- That's it.

For tests you could directly press Alt+R key (on Windows Device) while test.rb file is open.

That's all you need to know.

Vali-Mandeal commented:
Added proper description

On Tue, Oct 8, 2019 at 3:52 PM Calinoiu Alexandru Nicolae <
notifications@github.com> wrote:

> Please give a proper name to the pr and add a description to it with a
> minimal info on how to run the code, ruby version other deps ....
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Agilefreaks/test_oop/pull/16?email_source=notifications&email_token=AMKNPEY5ML5SZ666CPGZDELQNR7BRA5CNFSM4I6NQUKKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAUBONA#issuecomment-539498292>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMKNPE4PYK4TVUKGOCJ2U5DQNR7BRANCNFSM4I6NQUKA>
> .
>

alexandru-calinoiu commented on Main.rb:
```diff
@@ -0,0 +1,19 @@
+dirPath = File.dirname(File.dirname(File.expand_path(__FILE__)))
+load dirPath + "/CoffeeShopApp/Objects/CoffeeShop.rb"
```
When I try to run the code via

```
ruby Main.rb
```

I get the following error

Traceback (most recent call last):
        1: from Main.rb:2:in `<main>'
Main.rb:2:in `load': cannot load such file -- /<path>/CoffeeShopApp/Objects/CoffeeShop.rb (LoadError)
```

Please take a look at [require](https://ruby-doc.org/core-2.6.5/Kernel.html#method-i-require) and [require_relative](https://ruby-doc.org/core-2.6.5/Kernel.html#method-i-require_relative) kernel methods.

alexandru-calinoiu reviewed (changes requested):
Still very hard to review:

- Instruction on how to run should be in PR description not in a commit.
- See the comm that I left in the poorly named `Main.rb`

Every language has it's own idioms and coding styles, you have seemed to ignored them and this makes reviewing this impossible.

Please take a look at the https://www.rubocop.org/en/stable/ and focus on proper naming including filenames and variables.

alexandru-calinoiu commented:
Closing this for now, will followup via email.
